_search-plugins/star-tree-index.md

```diff
@@ -0,0 +1,175 @@
+  },
+  "mappings": {
+    "composite": {
+      "startree1": {
+        "type": "star_tree",
+        "config": {
+          "ordered_dimensions": [
+            {
+              "name": "status"
+            },
+            {
+              "name": "port"
+            }
+          ],
+          "metrics": [
+            {
+              "name": "request_size",
+              "stats": [
+                "sum",
+                "value_count",
+                "min",
+                "max"
+              ],
+              "name": "latency",
+              "stats": [
+                "sum",
+                "value_count",
+                "min",
+                "max"
+              ]
+            }
+          ]
+        }
+      }
+    },
+    "properties": {
+      "status": {
+        "type": "integer"
+      },
+      "port": {
+        "type": "integer"
+      },
+      "request_size": {
+        "type": "integer"
+      },
+      "latency": {
+        "type": "scaled_float",
+        "scaling_factor": 10
+      }
+    }
+  }
+}
+```
+
+For detailed information about Star Tree index mapping and parameters see [Star Tree field type]({{site.url}}{{site.baseurl}}/field-types/star-tree/).
+
+## Supported query and aggregations
+
+Star Tree index can be used to optimize aggregations for selected set of queries with support for more coming in upcoming releases.
+
+### Supported queries
+Ensure the following in star tree index mapping,
+- The fields present in the query must be present as part of `ordered_dimensions` as part of star-tree configuration.
+
+The following queries are supported [ when supported aggregations are specified ] <br/>
+
+- [Term query](https://opensearch.org/docs/latest/query-dsl/term/term/)
+- [Match all docs query](https://opensearch.org/docs/latest/query-dsl/match-all/)
+
+### Supported aggregations
+Ensure the following in star tree index mapping,
+- The fields present in the aggregation must be present as part of `metrics` as part of star-tree configuration.
+- The metric aggregation type must be part of `stats` parameter.
+ 
+Following metric aggregations are supported.
+- [Sum](https://opensearch.org/docs/latest/aggregations/metric/sum/)
+- [Minimum](https://opensearch.org/docs/latest/aggregations/metric/minimum/)
+- [Maximum](https://opensearch.org/docs/latest/aggregations/metric/maximum/)
+- [Value count](https://opensearch.org/docs/latest/aggregations/metric/value-count/)
+- [Average](https://opensearch.org/docs/latest/aggregations/metric/average/)
+
+### Examples
+To get sum of `request_size` for all error logs with `status=500` with the [example mapping](#defining-star-tree-index-in-mappings) :
+```json
+POST /logs/_search
+{
+  "query": {
+    "term": {
+      "status": "500"
+    }
+  },
+  "aggs": {
+    "sum_request_size": {
+      "sum": {
+        "field": "request_size"
+      }
+    }
+  }
+}
+```
+
+This query will get optimized automatically as star-tree index will be used.
```

Comment:
We can briefly mention the advantage of this being O(1) as the are just 2 lookups instead of scanning through all docs that are produced by the Lucene collector of query 

`TermQuery(Status=500)`